libs/x-content/impl/src/test/java/org/elasticsearch/xcontent/provider/json/ESUTF8StreamJsonParserTests.java

```diff
@@ -288,4 +298,116 @@ public void testGetValueRandomized() throws IOException {
         });
     }
 
+    /**
+     * This test compares the retrieval of an optimised text against the baseline.
+     */
+    public void testOptimisedParser() throws Exception {
+        for (int i = 0; i < 200; i++) {
+            String json = randomJsonInput(randomIntBetween(1, 6));
+            try (
+                XContentParser optimisedParser = TestXContentParser.create(json);
+                XContentParser baselineParser = TestXContentParser.create(json)
```

Comment:
Let's use a default parser as default:

```suggestion
                var baselineParser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, new BytesArray(json), XContentType.JSON)
```

and maybe initialize baselineParser first?